Load balancing backoffice documentation (#7545)
umbraco/UmbracoDocs: 17/umbraco-cms/reference/scheduling.md

### Implementing a custom distributed background job

To implement a custom distributed background job, create a class that implements the `IDistributedBackgroundJob` interface. Like with `IRecurringBackgroundJob` DI is available in the constructor.

```csharp
public class MyCustomBackgroundJob : IDistributedBackgroundJob
{
    private readonly ILogger<MyCustomBackgroundJob> _logger;
    public string Name => "MyCustomBackgroundJob";
    
    public TimeSpan Period { get; private set; }

    public MyCustomBackgroundJob(ILogger<MyCustomBackgroundJob> logger)
    {
        _logger = logger;
        Period = TimeSpan.FromSeconds(20);
    }
    
    public Task ExecuteAsync()
    {
        // Your custom background job logic here
        _logger.LogInformation("MyCustomBackgroundJob is executing.");
        return Task.CompletedTask;
    }
}
```

It's required to give your job a unique name via the `Name` property. This is used to track the job in the database.

> The period is specified via the `Period` property, which controls how often the job should be run, in this example it's every 20 seconds.

Reviewer: ```suggestion
The period is specified via the `Period` property, which controls how often the job should run. In this example, it runs every 20 seconds.
```
